enhanceWithPromiseSupport.js

```diff
@@ -0,0 +1,12 @@
+// if a callback is provided then don't resolve but use provided callback
+export const enhanceWithPromiseSupport = notification => ({
+    confirm: args =>
+        new Promise(resolve => {
+            notification.confirm({ callback: resolve, ...args });
+        }),
+    alert: args =>
+        new Promise(resolve => {
```

Comment:
can these never throw?

Reply:
They can.
What do you think of putting them in a try/catch and then rejecting when thrown?